### ISSUE TITLE
ERRINFO_GRAPHICS_MODE_NOT_SUPPORTED - Windows 10 Insider 17063

### ISSUE THREAD

**Issue by brhahlen** [FreeRDP/FreeRDP#4326]
Trying to connect to Windows Insider 17063 (just updated yesterday).
My connection, that used to work fine on 17046 and earlier, now starts connecting and then errors out with:
```
[09:20:29:674] [32327:7056] [INFO][com.freerdp.gdi] - Local framebuffer format  PIXEL_FORMAT_BGRA32
[09:20:29:674] [32327:7056] [INFO][com.freerdp.gdi] - Remote framebuffer format PIXEL_FORMAT_BGRA32
[09:20:29:728] [32327:7056] [INFO][com.freerdp.core] - ERRINFO_GRAPHICS_MODE_NOT_SUPPORTED (0x0000112D):The graphics mode requested by the client is not supported by the server.
[09:20:29:730] [32327:7056] [ERROR][com.freerdp.core.transport] - BIO_read returned a system error 0: Success
```
**Commandline used**
```remmina -c ~/.remmina/Faramir_ext.remmina &```
(Yes, it's remmina, but the problem seems to lie with freerdp).
**FreeRDP version**
```freerdp-libs.x86_64 2:2.0.0-34.20170831git3b83526.fc27```
**OS**
Fedora 27
**Remmina Config**
```
[remmina]
disableclipboard=0
ssh_auth=0
clientname=
quality=9
ssh_charset=
ssh_privatekey=
console=0
resolution=2560x1440
group=Home
password=.
name=Faramir_ext
ssh_loopback=0
shareprinter=0
ssh_username=
ssh_server=
security=
protocol=RDP
execpath=
sound=off
exec=
ssh_enabled=0
username=username@domain.com
sharefolder=/home/username
domain=
server=localhost:3389
colordepth=0
scale=1
aspectscale=1
hscale=0
window_maximize=0
viewmode=1
vscale=0
window_height=771
window_width=1362
keyboard_grab=0
sharesmartcard=0
cert_ignore=0
gateway_server=
gateway_usage=0
disablepasswordstoring=0
precommand=
loadbalanceinfo=
disableautoreconnect=0
postcommand=
microphone=0
resolution_height=1440
resolution_width=2560
gateway_password=
gateway_username=
gateway_domain=
```

**Comment by akallabeth** [comment]
Not a FreeRDP bug, MS decided to deactivate older RDP protocol features.
You need ```/gfx``` mode activated, then it will work.

**Comment by argonym** [comment]
Is `/gfx` still considered experimental? And BTW what's the difference between `/gfx:RFX` and `/rfx`.
Maybe it's time to enable some features that have been around for years by default.
If the server doesn't support RDP8 graphics, the FreeRDP would fall back to the protocol version the server supports, right?

**Comment by brhahlen** [comment]
How do I enable GFX mode?

**Comment by akallabeth** [comment]
@argonym yes, the client just announces what it supports. Same goes for the server and then a common feature set is used (mostly depending on the server policy settings)

**Comment by argonym** [comment]
Thank you for the info.
Though the xfreerdp man/help page seems contradicting to me as it suggest that only the `/gfx-h264` option is using RDP8.1. (Though that may be technically correct, I don't know.)

My suggestion to default to announce some more recent protocol features (maybe in nightly/preview/debug builds only) still stands.

**Comment by bmiklautz** [comment]
@akallabeth you got some pointer to the Microsoft documentation where they state something like this?
If it's really disabled by default we probably should make /gfx be enabled as default as well, I think.

**Comment by lemonmojo** [comment]
I can confirm this behavior in Windows 10 Insider build 17063. I've installed the build on a couple different machines, including VMs as well as real hardware and all of them require the graphics pipeline to successfully connect. Unfortunately with the graphical glitches when connecting to Windows 8 using FreeRDP I'm reluctant to enable this by default in my FreeRDP client until Microsoft actually releases this build to the public. Any news on this?

**Comment by hardening** [comment]
@lemonmojo do you have an issue in mind about glitches with egfx ?

**Comment by lemonmojo** [comment]
@hardening Check out this video: https://lemonmojo.com/temp/EGFX_Win8.mp4
This only happens with /gfx against Windows 8/8.1 machines. Windows 10 is perfectly fine.
Might be the same issue as https://github.com/FreeRDP/FreeRDP/issues/4266. @hardening also mentioned glitches [here](https://github.com/FreeRDP/FreeRDP/issues/4326#issuecomment-353576938).

**Comment by akallabeth** [comment]
@lemonmojo Do you have these issues with windows 8.1 too? Only could reproduce them with windows 8
Anyway, we could, of course, disable older GFX capabilities (anything before windows 10) and only announce them by default.

**Comment by lemonmojo** [comment]
@akallabeth The video was recorded when connected to a Windows 8.1 VM.

**Comment by Coffee-fan** [comment]
Passing +gfx-progressive on the freerdp command line worked for me when connecting to Windows 10 17074.

**Comment by lemonmojo** [comment]
FYI: I've just updated to build 17093 and am now again able to connect without enabling egfx.

**Comment by lemonmojo** [comment]
FYI: Build 17101 also doesn't require egfx.

**Comment by ET-Bent** [comment]
![image](https://user-images.githubusercontent.com/5788982/41921313-df8d901e-7962-11e8-89aa-aca3f753684b.png)
How do we use anything above `GFX RFX (32 bpp)` on Linux at all?

Ubuntu 18.04, Nvidia GTX 970 (nvidia proprietary drivers installed).

What do I need to install to get these modes working on the client side?

**Comment by akallabeth** [comment]
@ET-Bent a build with h264 enabled?

**Comment by ukos-git** [comment]
This is not related to the original issue but since the discussion was already started and I landed here searching for the same remmina error as @ET-Bent, I wanted to give some more information:

[standard remmina and also the ppa](https://launchpad.net/~freerdp-team/+archive/ubuntu/freerdp) are rather old. One should try the [the freerdp-nightly](https://github.com/FreeRDP/FreeRDP/wiki/PreBuilds) instead, as [suggested in the FAQ](https://github.com/FreeRDP/FreeRDP/wiki/FAQ#freerdp-shipped-with-my-distribution-is-rather-old---is-there-a-recent-version). It installs to `/opt/freerdp-nightly/bin/xfreerdp` on ubuntu which is probably not in the standard `$PATH` by default. Unfortunately it is built without gfx-mode enabled.
```bash
 /opt/freerdp-nightly/bin/xfreerdp --buildconfig | grep WITH_GFX_H264
```

I'm not sure why it is disabled by default.
